frontend/src/app/[locale]/opportunity/[id]/page.tsx

```diff
@@ -26,6 +27,7 @@ export async function generateMetadata({ params }: { params: { id: string } }) {
     title = `${t("OpportunityListing.page_title")} - ${opportunityData.opportunity_title}`;
   } catch (error) {
     console.error("Failed to render title");
+    return notFound();
```

Comment:
More a design question, but would we want to display a notFound page if the API returned a 5xx? Looks like

Reply:
Maybe a follow-up ticket to create a `500.js` error page and show that? https://nextjs.org/docs/pages/building-your-application/configuring/error-handling#handling-server-errors